core/src/saros/util/PathUtils.java

```diff
@@ -0,0 +1,141 @@
+  /**
+   * Removes the given number of segments from the end of the given path.
+   *
+   * <p>Returns an empty path if the number of segments to drop is larger or equal to the number of
+   * segments in the given path.
+   *
+   * @param path the path whose segments to drop
+   * @param count the number of segments to drop
+   * @return the remaining path
+   * @see Path#subpath(int, int)
+   */
+  public static Path removeLastSegments(Path path, int count) {
+    Objects.requireNonNull(path, "The given path must not be null");
+
+    if (path.getNameCount() <= count) {
+      return Paths.get("");
+    }
+
+    return path.subpath(0, path.getNameCount() - count);
+  }
+
+  /**
+   * Returns a platform-independent string representation of the given path. The given path must not
+   * be absolute.
+   *
+   * <p>The string can be converted into a path for the local filesystem using {@link
+   * PathUtils#fromPortableString(String)}.
+   *
+   * @param path the path to convert
+   * @return a platform-independent string representation of the given path
+   * @see PathUtils#fromPortableString(String)
+   * @throws IllegalArgumentException if the given path is absolute
+   */
+  public static String toPortableString(Path path) {
+    Objects.requireNonNull(path, "The given path must not be null");
+
+    if (path.isAbsolute()) {
+      throw new IllegalArgumentException("The given path must not be absolute");
+    }
+
+    String localPathString = path.toString();
+
+    if (File.separator.equals(PORTABLE_SEPARATOR)) {
+      if (localPathString.contains(WINDOWS_SEPARATOR)) {
+        log.warn(
+            "The given local path contains occurrences of the Windows separator character '"
+                + WINDOWS_SEPARATOR
+                + "'. As this character is not seen as a separator by the local filesystem, this "
+                + "will most likely lead to issues when the path is transferred to a Saros "
+                + "instance running on a Windows system.");
+      }
+
+      return localPathString;
+    }
+
+    return localPathString.replace(File.separator, PORTABLE_SEPARATOR);
+  }
+
+  /**
+   * Returns a path in the local filesystem for the given platform-independent path string. This
+   * method is only intended to work with strings generated by {@link
+   * PathUtils#toPortableString(Path)}.
+   *
+   * @param portablePathString the path string to convert
+   * @return a path in the local filesystem for the given platform-independent path string
+   * @see PathUtils#toPortableString(Path)
+   */
+  public static Path fromPortableString(String portablePathString) {
+    Objects.requireNonNull(portablePathString, "The given string must not be null");
+
+    if (File.separator.equals(PORTABLE_SEPARATOR)) {
+      return Paths.get(portablePathString);
+    }
+
+    if (portablePathString.contains(WINDOWS_SEPARATOR)) {
+      log.warn(
+          "The given portable path contains occurrences of the Windows separator character '"
```

Comment:
I will add the path to the log messages and adjust the content.

Regarding the issue of the current caller, I am not really sure what to do in that regard. I could throw an exception in this case or move this check to a separate method so that the caller would have to deal with it, but I am not whether that would be a better option.